src/dashboard/src/components/api/views.py

```diff
@@ -48,26 +55,190 @@ def authenticate_request(request):
+        return django.http.HttpResponseNotAllowed(['GET'])
+    auth_error = authenticate_request(request)
+    response = {}
+    if auth_error is not None:
+        response = {'message': auth_error, 'error': True}
+        return django.http.HttpResponseForbidden(
+            json.dumps(response),
+            mimetype='application/json'
+        )
+    error = None
+
+    # Get info about unit
+    if unit_type == 'unitTransfer':
+        unit = get_object_or_None(models.Transfer, uuid=unit_uuid)
+        response['type'] = 'transfer'
+    elif unit_type == 'unitSIP':
+        unit = get_object_or_None(models.SIP, uuid=unit_uuid)
+        response['type'] = 'SIP'
+
+    if unit is None:
+        response['message'] = 'Cannot fetch {} with UUID {}'.format(unit_type, unit_uuid)
+        response['error'] = True
+        return django.http.HttpResponseBadRequest(
+            json.dumps(response),
+            mimetype='application/json',
+        )
+    directory = unit.currentpath if unit_type == 'unitSIP' else unit.currentlocation
+    response['path'] = directory.replace('%sharedPath%', SHARED_DIRECTORY_ROOT, 1)
+    response['directory'] = os.path.basename(os.path.normpath(directory))
+    response['name'] = response['directory'].replace('-' + unit_uuid, '', 1)
+    response['uuid'] = unit_uuid
+
+    # Get status (including new SIP uuid, current microservice)
+    status_info = get_unit_status(unit_uuid, unit_type)
+    response.update(status_info)
+
+    if error is not None:
+        response['message'] = error
+        response['error'] = True
+        return django.http.HttpResponseServerError(
+            json.dumps(response),
+            mimetype='application/json'
+        )
+    response['message'] = 'Fetched status for {} successfully.'.format(unit_uuid)
+    return helpers.json_response(response)
+
+
+def waiting_for_user_input(request):
+    # Example: http://127.0.0.1/api/transfer/waiting?username=mike&api_key=<API key>
+    if request.method not in ('GET',):
+        return django.http.HttpResponseNotAllowed(['GET'])
+
+    auth_error = authenticate_request(request)
+    response = {}
+    if auth_error is not None:
+        response = {'message': auth_error, 'error': True}
+        return django.http.HttpResponseForbidden(
+            json.dumps(response),
+            mimetype='application/json'
+        )
+
+    error = None
+    waiting_units = []
+
+    jobs = models.Job.objects.filter(currentstep='Awaiting decision')
+    for job in jobs:
+        unit_uuid = job.sipuuid
+        directory = os.path.basename(os.path.normpath(job.directory))
+        unit_name = directory.replace('-' + unit_uuid, '', 1)
+
+        waiting_units.append({
+            'sip_directory': directory,
+            'sip_uuid': unit_uuid,
+            'sip_name': unit_name,
+            'microservice': job.jobtype,
+            # 'choices': []  # TODO? Return list of choices, see ingest.views.ingest_status
```

Comment:
Probably not.  I thought having the list of choices would be a useful tidbit to send back, but ingest_status does ridiculous contortions to get the info, and I decided it wasn't worthwhile until something needed it.